### DISCUSSION TITLE
How to tunnel status port over ngrok?

### DISCUSSION THREAD

**Discussion by aaronadamsCA** [gatsbyjs/gatsby#28443] [Help] [answered]
I'm developing a Shopify admin app using Gatsby and ngrok. The dev flow looks like this:

- `gatsby develop` serves site at `http://localhost:8000`
- ngrok exposes `http://localhost:8000` at `https://my-app.ngrok.io`
- Shopify iframes `https://my-app.ngrok.io` and the app is accessed via the Shopify admin

I don't think there's any way to tunnel the status messages, since they run on a different port, but ngrok doesn't support exposing other ports over the Internet. I could tunnel the status port to another hostname, but I don't see how that gets me any closer to this working.

Ideally Socket.io would listen on the same port as the development server. Maybe that's just a v3 feature request?

**Comment by KyleAMathews** [comment, marked as answer]
Not necessarily a v3 feature request — agree that ideally we're just using one port — @mxstbr did you ever look into proxying the status server through the regular dev server?

**Reply by aaronadamsCA** [reply]
Sorry @mxstbr, I hope you don't mind a polite bump on this. I'm just wondering if there's an issue I can follow/file regarding moving Socket.io to the same port as Gatsby. Thanks!

**Reply by mxstbr** [reply]
I did not look into this! I assume it would be possible but requires some finessing with the websocket stuff.

**Reply by aaronadamsCA** [reply]
Filed #31898, thanks!
